x-pack/plugins/cases/public/components/configure_cases/index.tsx

```diff
@@ -264,21 +268,47 @@ export const ConfigureCases: React.FC = React.memo(() => {
     ]
   );
 
+  const onEditCustomField = useCallback(
+    (key: string) => {
+      const selectedCustomField = customFields.find((item) => item.key === key);
+
+      if (selectedCustomField) {
+        setCustomFieldToEdit(selectedCustomField);
+      }
+      setCustomFieldFlyoutVisibility(true);
+    },
+    [setCustomFieldFlyoutVisibility, setCustomFieldToEdit, customFields]
+  );
+
   const onCloseAddFieldFlyout = useCallback(() => {
-    setAddFieldFlyoutVisibility(false);
-  }, [setAddFieldFlyoutVisibility]);
+    setCustomFieldFlyoutVisibility(false);
+    setCustomFieldToEdit(null);
+  }, [setCustomFieldFlyoutVisibility, setCustomFieldToEdit]);
 
-  const onCustomFieldCreated = useCallback(
+  const onSaveCustomField = useCallback(
     (customFieldData: CustomFieldConfiguration) => {
+      let updatedFields: CustomFieldsConfiguration = [];
+      const isExistingCustomField = customFields.find((field) => field.key === customFieldData.key);
+
+      if (isExistingCustomField) {
+        updatedFields = customFields.map((item) => {
+          if (item.key === customFieldData.key) {
+            return customFieldData;
+          }
+          return item;
+        });
+      }
```

Comment:
There is a helper function you can use in `x-pack/plugins/cases/public/components/case_view/components/custom_fields.tsx` called `addOrReplaceCustomField`. You can move it to `components/custom_fields/utils.ts` (does not exist) or `x-pack/plugins/cases/public/components/utils.ts` and change the type of the function to accept `customFields: CaseUICustomField[] | CustomFieldConfiguration[]`